src/types.ml

```diff
@@ -750,17 +767,17 @@ type 'a const =
   | Const_struct
 
 let pp_const fmt = function
-  | Const_I32 i -> pp fmt "i32.const %ld" i
-  | Const_I64 i -> pp fmt "i64.const %Ld" i
-  | Const_F32 f -> pp fmt "f32.const %a" Float32.pp f
-  | Const_F64 f -> pp fmt "f64.const %a" Float64.pp f
-  | Const_null rt -> pp fmt "ref.null %a" pp_heap_type rt
-  | Const_host i -> pp fmt "ref.host %d" i
-  | Const_extern i -> pp fmt "ref.extern %d" i
+  | Const_I32 i -> pp fmt "(i32.const %ld)" i
+  | Const_I64 i -> pp fmt "(i64.const %Ld)" i
+  | Const_F32 f -> pp fmt "(f32.const %a)" Float32.pp f
+  | Const_F64 f -> pp fmt "(f64.const %a)" Float64.pp f
+  | Const_null rt -> pp fmt "(ref.null %a)" pp_heap_type rt
+  | Const_host i -> pp fmt "(ref.host %d)" i
+  | Const_extern i -> pp fmt "(ref.extern %d)" i
   | Const_array -> pp fmt "ref.array"
   | Const_eq -> pp fmt "ref.eq"
   | Const_i31 -> pp fmt "ref.i31"
   | Const_struct -> pp fmt "ref.struct"
```

Comment:
The parens are not needed for `ref.{array,eq,i31,struct}` ?

Reply:
I don't know,
since `ref.{array,eq,i31,struct}` are just one "word" it is probably not needed, I dont' know how to check this

Reply:
These are used to test the return value of an invoke through `assert_return` but they're from the GC proposal so you'd need to clone [WebAssembly/gc](https://github.com/WebAssembly/gc) to try them.

You can find some files in `test/gc` (they'll be moved to `test/reference/gc` once we support the GC proposal completely).

If you look at the `i31.wast` file, you'll see one instance of this (line 28).

The reason they have a single work is because we can not know what will be the real value hold by the reference in case of struct/eq/array etc. (we could for i31 but that's another story).

Anyway, I tried removing the parenthesis and got an error from the reference interpreter so I guess they should stay here.